service/tbc/tbc.go

```diff
@@ -1469,37 +1515,46 @@ func (s *Server) handleInv(ctx context.Context, p *peer, msg *wire.MsgInv, raw [
 	log.Tracef("handleInv (%v)", p)
 	defer log.Tracef("handleInv exit (%v)", p)
 
+	var txsFound bool
+
 	for _, v := range msg.InvList {
 		switch v.Type {
 		case wire.InvTypeError:
 			log.Errorf("inventory error: %v", v.Hash)
 		case wire.InvTypeTx:
-			// XXX add to mempool
-			// log.Infof("inventory tx: %v", v.Hash)
+			// handle these later or else we have to insert txs one
+			// at a time while taking a mutex.
+			txsFound = true
 		case wire.InvTypeBlock:
-			log.Infof("inventory block: %v", v.Hash)
+			log.Debugf("inventory block: %v", v.Hash)
 		case wire.InvTypeFilteredBlock:
-			log.Infof("inventory filtered block: %v", v.Hash)
+			log.Debugf("inventory filtered block: %v", v.Hash)
 		case wire.InvTypeWitnessBlock:
 			log.Infof("inventory witness block: %v", v.Hash)
 		case wire.InvTypeWitnessTx:
 			log.Infof("inventory witness tx: %v", v.Hash)
 		case wire.InvTypeFilteredWitnessBlock:
-			log.Infof("inventory filtered witness block: %v", v.Hash)
+			log.Debugf("inventory filtered witness block: %v", v.Hash)
 		default:
-			log.Infof("inventory unknown: %v", spew.Sdump(v.Hash))
+			log.Errorf("inventory unknown: %v", spew.Sdump(v.Hash))
+		}
+	}
+
+	if s.cfg.MempoolEnabled && txsFound {
+		if err := s.mempool.invTxsInsert(ctx, msg); err != nil {
+			go s.downloadMissingTx(ctx, p)
```

Comment:
is the _only error we'll ever get from invTxsInsert_ a missing tx? it may be the case now, but this may be a gotcha if we end up returning different errors from `invTxsInsert`.  Can we do a check that the err is indeed "tx missing"?

Reply:
That is correct and this code is ossified.